.github/workflows/tools.yaml

```
@@ -0,0 +1,18 @@
+name: OpenAPI Tools build
+on:
+  pull_request:
+    paths:
+      - "tools/**"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - working-directory: ./tools
+        run: |
+          npm install
+          npm ci
```

are you sure this is running your `ci` script instead of the `npm-ci - Clean install a project` command?
I think for your script you may need to do 
```suggestion
          npm run ci
```